### DISCUSSION TITLE
[Idea]: Portable version for Windows

### DISCUSSION THREAD

**Discussion by ux-git** [sircharlo/meeting-media-manager#3199] [Ideas] [answered]
### Is your idea related to a problem?

Do you consider creating a portable version for Windows? MacOS does already have this on the system level.

### Describe your idea

A USB-stick-friendly version might be quite useful for emergency help for those, who don't use M³. Since we've recently faced plenty of JW Library bugs, this quick solution could have saved us many times.

1.Connect your personal USB stick, 2. Open the app, 3. Sync the media, 4. Present them. Only two minutes to organize all this.

Emergency tool 👨‍⚕️

### Describe alternatives you've considered

Bringing a laptop to the meeting and connecting it to an external HDMI input

**Comment by mtdvlpr** [comment]
How do you envision this feature? Would the congregation settings be saved on the usb stick? Would the additional media be saved on the usb stick? Would the cache of jwpub files be stored on the usb stick?

**Reply by ux-git** [reply]
Yes, most likely all the data and settings should be stored on the stick to ensure real portability.

This use case is for those who don't use M³, therefore it's not installed (yet). So inserting the USB drive, saving the meeting and then demonstrating the app for others is quite realistic, though rare, scenario. It's definitely not for weekly use, therefore storing settings and some jwpub cache + some media shouldn't be a bis problem. For the portable version it makes sense to have some kind of auto-clear period for old caches, since the drive's capacity is quite limited.

But that's basically it. Even if the current version with no changes could be simply installed on a drive, it would be wonderful.

**Comment by sircharlo** [comment]
A "portable" version exists [here](https://github.com/sircharlo/meeting-media-manager/actions/runs/11921366463), but it's only portable in the sense that it doesn't require installation. We'd have to look into making it fully portable, as in only touching the directory the executable file is located in.

**Reply by ux-git** [reply]
Thanks! Could you please clarify what does the following mean? 

>  it's only portable in the sense that it doesn't require installation

What other aspects of portability are missing? What is further planned?

**Reply by sircharlo** [reply, marked as answer]
A true portable app only writes to its own directory and leaves no trace that it was ever run. As it was, the app would open without installation, but would write its settings and downloaded media cache to the Windows AppData directory just like the regular app. 

I've since fixed that as of https://github.com/sircharlo/meeting-media-manager/pull/3415/commits/a997bae5e57bbe069f840733f7f601661433d83f, though. 😄 
The next version will include a (truly) portable version for those who know to look for it!

**Reply by ux-git** [reply]
Wow! Awesome, you're fast! 

I'll be able to test the portable version tomorrow, I'm excited to give it a try 🙂

**Reply by sircharlo** [reply]
Let me know if it works as intended!

**Reply by ux-git** [reply]
@sircharlo what is a workaround to determine and clear the temp folder after using the current portable version?

**Reply by sircharlo** [reply]
No need to clear anything. It just reused the existing installation's settings and folders. 
The latest portable build [here](https://github.com/sircharlo/meeting-media-manager/actions/runs/11922394674) should create a new folder wherever the `exe` is located to store the portable data. For example, if the portable application is on a flash drive, the directory with cache and publications will be right next to it.

**Reply by mtdvlpr** [reply]
Look who's on fire now 🔥

**Reply by ux-git** [reply]
@sircharlo 
![IMG_20241120_104715_290](https://github.com/user-attachments/assets/96a4e695-870c-4be4-99fb-91dce5bdf34f)

**Reply by mtdvlpr** [reply]
@ux-git, was this immediately after opening or did you perform a specific action?

**Reply by ux-git** [reply]
@mtdvlpr yeah, it simply won't open

**Reply by mtdvlpr** [reply]
Hmm, I don't have this problem. Could you try the latest build?

https://github.com/sircharlo/meeting-media-manager/actions/runs/11931453279

**Reply by ux-git** [reply]
Ok, I'll give it a try. Is there anything that must be installed on a computer in order to run the app?

**Reply by sircharlo** [reply]
Nothing at all. I downloaded it and ran it from my  Downloads folder; a new folder was automatically created in Downloads for the app's data and everything works as intended for me.

**Reply by ux-git** [reply]
@sircharlo if we're talking about running from a USB stick, is there any limitations or specific formatting format that user should be aware of?

**Reply by sircharlo** [reply]
No; as long as the directory the portable executable is located in is writable, everything _should_ be smooth.

**Reply by sircharlo** [reply]
> Ok, I'll give it a try. Is there anything that must be installed on a computer in order to run the app?

Oh I think I know what happened! Did you run the portable app from the zip file directly, without unzipping it first? (The test builds are zipped, contrary to the production ones.)
To test properly, copy the portable executable (just the exe file) to a folder or flash drive and try opening the portable app after.

**Reply by ux-git** [reply]
I definitely unzipped it. Downloaded and unzipped in Downloads folder and then moved to the USB drive and ran from it. I'll try to see how it behaves in different scenarios and will let you know. Perhaps something is wrong about the USB stick itself

**Reply by ux-git** [reply]
@sircharlo the app works when I start it from the computer's Downloads folder. Currently, I can't start it from the flash drive. I'll continue testing and will keep you updated what was (currently is) wrong.

**Reply by mtdvlpr** [reply]
Given the error message, it might help if you place the portable app in a subfolder on the usb.

**Reply by sircharlo** [reply]
> @sircharlo the app works when I start it from the computer's Downloads folder. Currently, I can't start it from the flash drive. I'll continue testing and will keep you updated what was (currently is) wrong.

Interesting. I'll test on my side too.

**Reply by ux-git** [reply]
@mtdvlpr yes, it's exactly the issue. Using the latest version suggested by you, I managed to open it only from a subfolder. After moving it to the root directory, the same file won't open and has the same error.

**Reply by ux-git** [reply]
The portable version works really good so far! I only noticed that the media playback is turned off by default, if the setup wizard was skipped (I somehow managed to open the portable version from Downloads and then after running from the USD drive it didn't start with the wizard, or I mixed something up).

**Reply by ux-git** [reply]
Plus the JW Library companion mode was on by default.

And is there any way to switch the congregation profile type to "Other" **after** it's been created? Can't find such setting.

**Reply by mtdvlpr** [reply]
> Plus the JW Library companion mode was on by default.
> 
> And is there any way to switch the congregation profile type to "Other" **after** it's been created? Can't find such setting.

That is the "JW Library companion mode". When that is turned on, you're using the "Other" profile. The setting might need to be renamed though to better signify that.

**Reply by sircharlo** [reply]
Probably be good to change that to "Profile Type", with an explanation, and hide all the irrelevant settings in that case.

**Reply by sircharlo** [reply]
> Probably be good to change that to "Profile Type", with an explanation, and hide all the irrelevant settings in that case.

Step 1: Irrelevant settings hidden when profile type is Other as of https://github.com/sircharlo/meeting-media-manager/commit/6acf15fd904cfbcf1c8b775079904a7ea8da062f.

**Reply by sircharlo** [reply]
What do you think of this?

---
![image](https://github.com/user-attachments/assets/e569f277-81f6-41c2-ad54-9d36a23fbd45)
